buildengine/engine.go

```diff
@@ -212,8 +238,55 @@ func (e *Engine) watchForModuleChanges(ctx context.Context, period time.Duration
 					logger.Errorf(err, "deploy %s failed", event.Module.Module)
 				}
 			}
+		case change := <-schemaChanges:
```

Comment:
The design we discussed was that the `WatchEventModuleChanged` event would just have a hash added to it. Then you just use the normal watch system to detect if it's a schema change or a code change.

Reply:
I started down that path, but since `Watch` was just watching for filesystem changes in the directories it seemed like it would need to do more than that to signal actual schema changes. I'd love to make that work though if that approach is better. The reason I did it this way is because I can make change to the code that would update the hash, but they aren't actual schema changes. For example if I add some new helper funcs or non-schema things, it would trigger us to rebuild the other modules. Whereas, really I don't think we need to rebuild other modules if the underlying schema does change.

I'm might be overthinking it though 🤔

Reply:
Ah that makes sense. Because we wouldn't have the compiled schema in `Watch()`, only files?

Reply:
That was my thought for now, but def open to ideas here.